gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -136,54 +136,19 @@ MacroExpander::expand_invoc (AST::MacroInvocation &invoc)
 
   // lookup the rules for this macro
   NodeId resolved_node = UNKNOWN_NODEID;
-  bool found = resolver->get_macro_scope ().lookup (
-    Resolver::CanonicalPath::new_seg (invoc.get_pattern_node_id (),
-				      invoc_data.get_path ().as_string ()),
-    &resolved_node);
-  if (!found)
-    {
-      rust_error_at (invoc.get_locus (), "unknown macro 1");
-      return;
-    }
-
-  // lookup the rules
-  AST::MacroRulesDefinition *rules_def = nullptr;
-  bool ok = mappings->lookup_macro_def (resolved_node, &rules_def);
-  rust_assert (ok);
-
-  auto fragment = AST::ASTFragment::create_error ();
-
-  if (rules_def->is_builtin ())
-    fragment
-      = rules_def->get_builtin_transcriber () (invoc.get_locus (), invoc_data);
+  NodeId source_node = UNKNOWN_NODEID;
+  if (has_semicolon)
+    source_node = invoc.get_macro_node_id ();
   else
-    fragment
-      = expand_decl_macro (invoc.get_locus (), invoc_data, *rules_def, false);
-
-  set_expanded_fragment (std::move (fragment));
-}
-
-// FIXME: Arthur: Refactor these two functions, they're really similar
```

Comment:
Are you looking for this `FIXME` comment? 😄

Reply:
Perfect hahaha, thank you :)